go.mod

```diff
@@ -33,22 +33,18 @@ require (
 	github.com/facebookgo/httpcontrol v0.0.0-20150708234001-ccde4420e1fe
 	github.com/facebookgo/stack v0.0.0-20160209184415-751773369052 // indirect
 	github.com/facebookgo/subset v0.0.0-20150612182917-8dac2c3c4870 // indirect
-	github.com/fortytw2/leaktest v1.3.0 // indirect
 	github.com/garyburd/redigo v0.0.0-20151219232044-836b6e58b335
 	github.com/glendc/gopher-json v0.0.0-20170414221815-dc4743023d0c // indirect
 	github.com/gocarina/gocsv v0.0.0-20190927101021-3ecffd272576 // indirect
 	github.com/godbus/dbus v4.0.1-0.20160727174541-7a8c533d28e8+incompatible // indirect
 	github.com/golang/freetype v0.0.0-20150924013838-f29eb116deb3 // indirect
-	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b // indirect
-	github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4
-	github.com/golang/protobuf v1.2.1-0.20190109072247-347cf4a86c1c // indirect
+	github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
 	github.com/gomodule/redigo v2.0.0+incompatible // indirect
 	github.com/google/cadvisor v0.21.1-0.20160212224345-e9739af18411
 	github.com/google/go-github v0.0.0-20151126072848-44b1ede22d71
 	github.com/google/go-querystring v0.0.0-20151028211038-2a60fc2ba6c1 // indirect
-	github.com/gorilla/context v1.1.1 // indirect
 	github.com/gorilla/css v0.0.0-20150317222238-a80e24ada269 // indirect
-	github.com/gorilla/mux v0.0.0-20160920230813-757bef944d0f
+	github.com/gorilla/mux v1.6.2
```

Comment:
Why has this dependency changed? `1.6.2` is almost two years newer than the previously pinned version and I can't see it being a direct dependency of any of the elastic modules? Similar for the other dependency changes - not saying that they're not worth bumping, just wondering why that happened here?

Reply:
Thank you very much for your code review!

This dependency upgrade is unintended.
In ES 7 client go.mod(https://github.com/olivere/elastic/blob/v7.0.9/go.mod#L14-L15), it lists `go.opencensus.io v0.22.1` as direct dependency and `google.golang.org/api v0.3.1` as indirect dependency, which isn't right. I submitted an issue here(https://github.com/olivere/elastic/issues/1266).

`google.golang.org/api v0.3.1` also requires `go.opencensus.io`, but it requires an old version v0.20.1. https://github.com/googleapis/google-api-go-client/blob/v0.3.1/go.mod

For `go.opencensus.io` module itself, the selected version is the higher version v0.22.1.

There's dramatic dependency change between v0.20.1 and v0.22.1.
https://github.com/census-instrumentation/opencensus-go/commit/ed3a3f0bf00d34af1ca7056123dae29672ca3b1a#diff-37aff102a57d3d7b797f152915a6dc16
In v0.20.1, it has several exporter related packages, like zipkin-go which uses mux v1.6.2: https://github.com/openzipkin/zipkin-go/blob/v0.1.6/go.mod#L14

Before Go modules decides which version to select, it needs to construct the complete build list first. That's to say, it appends dependencies of both v0.20.1 and v0.22.1 even if v0.20.1 is not selected eventually.

To summarize, if ES client prunes `google.golang.org/api v0.3.1` from its go.mod, there's no version bumping for `github.com/gorilla/mux`, `github.com/prometheus/client_golang`, `github.com/prometheus/common` etc.

Reply:
Looks like this is now fixed in the upstream elastic client
https://github.com/olivere/elastic/commit/5d20c799557a7637d81468010fc1246cef0b8ec6

Reply:
Updated the go.mod and go.sum based on the newest version and it's no longer pulling in the unneeded deps